### ISSUE TITLE
Add example to readme showing multiple sets of groups

### ISSUE THREAD

**Issue by 501st-alpha1** [alphapapa/org-super-agenda#22]
I just noticed that when I am in an `org-super-agenda`-enabled buffer, and I try to refresh it with `r` or `g`, it loses the super agenda enhancements, and goes back to a plain agenda.

This occurs at commit dc4dc51, which is the latest that I'm able to test due to #21, but I don't see anything in the commit messages since then that indicates this has changed.

Is this expected, and can it be fixed?  It would be nice to not have to reopen the agenda view every time.

**Comment by alphapapa** [comment]
Hi,

It works for me.  I had this issue in the past, but somehow I got it working.  I think it depends on how you have things configured.

However, it might also be fixed in the newer versions that you're unable to use because of using Emacs 24.4.  I don't remember the details now.  If you can upgrade to Emacs 25, the problem might go away, and if not, I can help you with configuration.

**Comment by 501st-alpha1** [comment]
Okay, I'll report back once I've upgraded.

**Comment by alphapapa** [comment]
Ping?  :)

**Comment by novoid** [comment]
I can confirm the issue with using Emacs 25.1.1 and org-super-agenda from git 89abc87 (4 months ago).

Then I did a git pull to the most current version (d06bfb0) and could reproduce this issue again.

My configuration file: https://github.com/novoid/dot-emacs/blob/master/config.org#my-super-agenda

**Comment by alphapapa** [comment]
@novoid This seems to be because you're binding `org-super-agenda-groups` in the `let` rather than in global scope, so it's only bound for the first time the command is run.  So, instead of using `(setq my-super-agenda-groups ...`, just do `(setq org-super-agenda-groups...`.

**Comment by novoid** [comment]
Hi,

Thanks for the detailed explanation and fixed config.

Unfortunately, this was not working out (`eval: Wrong type argument: listp, "a"`).

Doesn't really matter for me since I got used to kill and re-generate instead of just re-fresh my agenda.

The relevant config snippet related to super-agenda looked like that in case you really want to take a closer look again: http://paste.grml.org/3019/

**Comment by alphapapa** [comment]
@novoid There is no call to `eval` in org-super-agenda, so that error must be caused by something else.  If you provide a backtrace with `toggle-debug-on-error`, maybe we can track it down.

**Comment by 501st-alpha1** [comment]
@alphapapa That was indeed the cause of my issue, thanks!

This does lead to another issue, however.  The reason I had been defining `org-super-agenda-groups` in a `let` block is because I have different sets of groups for different purposes (mainly two right now, which I mentioned in #2).  So if I view one agenda with group set A, and another agenda with group set B, then when I refresh the first agenda, it loses the grouping (and is instead grouped according to set B, where possible).

Would it be possible to bind `org-super-agenda-groups` a single agenda/buffer to support this mode of operation?  (Let me know if you'd rather I open a new issue for this request.)

**Comment by alphapapa** [comment]
@501st-alpha1 That's interesting, I hadn't thought about using two sets of groups.

I guess you could bind `org-super-agenda-groups` buffer-locally after the agenda buffer is created, so something like:
```el
(let ((org-super-agenda-groups (list ...)))
  (org-agenda ...)
  (setq-local org-super-agenda-groups org-super-agenda-groups))
```
And then refreshing that agenda buffer should use those groups.  However, I think this code isn't quite enough, because you need to make sure that the `setq-local` is run inside the new agenda buffer.  So I guess you need to make a list of all existing agenda buffers, then open the new agenda, then figure out which agenda buffer is the new one, and then run the `setq-local` inside that buffer.  I think that will do it.

**Comment by 501st-alpha1** [comment]
@alphapapa That snippet works great, thanks!

> However, I think this code isn't quite enough, because you need to make sure that the `setq-local` is run inside the new agenda buffer.

I didn't have to do anything else here.  By default (at least in my Emacs), calling `org-agenda` automatically switches to the new agenda buffer, so `setq-local` is already run in the correct place.

Could this snippet be added to the README or examples, so other users can find it without digging through the issues?

**Comment by alphapapa** [comment]
> I didn't have to do anything else here. By default (at least in my Emacs), calling org-agenda automatically switches to the new agenda buffer, so setq-local is already run in the correct place.

That's great news!  That makes it much simpler.  I didn't remember that it switches to the agenda buffer first.

> Could this snippet be added to the README or examples, so other users can find it without digging through the issues?

Absolutely!  I'm a bit busy right now, but I'll get to it...  :)

**Comment by 501st-alpha1** [comment]
Addendum: the above snippet worked for the first refresh, but I later noticed that after that, the local value was lost and the behavior reverted on the next refresh.  I couldn't find anything in org-super-agenda that might be causing that, so I started looking at the org-agenda code, when I noticed `org-agenda-local-vars`, which is documented as:

> Variables that must be local in agenda buffers to allow multiple buffers.

So I did:

```el
(add-to-list 'org-agenda-local-vars 'org-super-agenda-groups)
```

and that fixed the issue.

And actually, I removed the `setq-local` line, and I was still able to have two different agendas with different groups, so it looks like my above config line is not only a necessary, but a sufficient condition to get this working.  So rather than add the example to the README, can the above line be added directly to the code?
